modules/src/mock/context.rs

```diff
@@ -991,6 +1016,16 @@ impl ClientReader for MockContext {
         self.latest_height
     }
 
+    fn host_consensus_state(&self, height: Height) -> Result<AnyConsensusState, Ics02Error> {
+        if height.is_zero() {
+            return Ok(self.pending_block.clone().into());
```

Comment:
We should avoid using `Height::zero()` as the magic parameter for getting the latest height. If that is needed, it is better to define a separate method `latest_host_consensus_state`. There is also no need for the `pending_block` here. If it is needed, it looks like the mock context should be able to deterministically generate the next block, so no caching should be needed.

I also don't see `host_consensus_state(Height::zero())` being used anywhere other than getting the host timestamp. So perhaps this is not needed if `host_timestamp` can be improved as suggested.

Reply:
Done! 👍 https://github.com/informalsystems/ibc-rs/pull/1771/commits/4e4847eb7bb26b7f53b9934591ef7fb8a4d2d5bc